src/Fixer/FunctionNotation/PhpdocToParamTypeFixer.php

```diff
@@ -0,0 +1,326 @@
+                ->setAllowedTypes(['bool'])
+                ->setDefault(true)
+                ->getOption(),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; 0 < $index; --$index) {
+            if (!$tokens[$index]->isGivenKind(T_FUNCTION)) {
+                continue;
+            }
+
+            $funcName = $tokens->getNextMeaningfulToken($index);
+            if ($tokens[$funcName]->equalsAny($this->blacklistFuncNames, false)) {
+                continue;
+            }
+
+            $paramTypeAnnotations = $this->findParamAnnotations($tokens, $index);
+
+            foreach ($paramTypeAnnotations as $paramTypeAnnotation) {
+                $types = array_values($paramTypeAnnotation->getTypes());
+                $typesCount = \count($types);
+                if (1 > $typesCount || 2 < $typesCount) {
+                    continue;
+                }
+
+                $isNullable = false;
+                $paramType = current($types);
+                if (2 === $typesCount) {
+                    $null = $types[0];
+                    $paramType = $types[1];
+                    if ('null' !== $null) {
+                        $null = $types[1];
+                        $paramType = $types[0];
+                    }
+
+                    if ('null' !== $null) {
+                        continue;
+                    }
+
+                    $isNullable = true;
+
+                    if (\PHP_VERSION_ID < 70100) {
+                        continue;
+                    }
+
+                    if ('void' === $paramType) {
+                        continue;
+                    }
+                }
+
+                if (isset($this->skippedTypes[$paramType])) {
+                    continue;
+                }
+
+                if (isset($this->versionSpecificTypes[$paramType]) && \PHP_VERSION_ID < $this->versionSpecificTypes[$paramType]) {
+                    continue;
+                }
+
+                if (1 !== Preg::match($this->classRegex, $paramType, $matches)) {
+                    continue;
+                }
+
+                if (isset($matches['array'])) {
+                    $paramType = 'array';
+                }
+
+                $startIndex = $tokens->getNextTokenOfKind($index, ['(']) + 1;
+                $variableIndex = $this->findCorrectVariable($tokens, $startIndex - 1, $paramTypeAnnotation);
+                if (null === $variableIndex) {
+                    continue;
+                }
+
+                if ('(' === $tokens[$variableIndex - 1]->getContent()) {
+                    if (!$tokens[$variableIndex]->isGivenKind([T_VARIABLE])) {
+                        continue;
+                    }
+                } else {
+                    if ($this->hasParamTypeHint($tokens, $variableIndex - 2)) {
```

Comment:
I think there is no test case covering this scenario.

Reply:
You refactored (and changed) these `if` conditions but you didn't add any test so I think the scenario is still not covered. Please add a test case that proves the typehint isn't changed if already defined, e.g.:
```php
'typehint already defined' => [
    '<?php /** @param string $foo */ function foo(int $foo) {}'
],
```